ores/utilities/stress_test.py

```diff
@@ -0,0 +1,168 @@
+    url_cycle = cycle(ores_urls)
+    rev_ids_cycle = cycle(rev_ids)
+    stats = defaultdict(int)
+    stats["requests"] = defaultdict(int)
+    stats["responses"] = defaultdict(int)
+    stats["model_scored"] = defaultdict(int)
+    stats["model_errored"] = defaultdict(lambda: defaultdict(int))
+
+    start = time.time()
+    with ThreadPoolExecutor(max_workers=REQUEST_THREADS) as executor:
+        try:
+            while time.sleep(delay) is None:
+                ores_url = next(url_cycle)
+                rev_id_batch = list(islice(rev_ids_cycle, batch_size))
+                logger.debug("Submitting {0} rev_ids to {1}...".format(
+                    len(rev_ids), ores_url))
+                executor.submit(request_and_stat, ores_url, context,
+                                models, rev_id_batch, stats, verbose)
+        except KeyboardInterrupt:
+            print("Caught keyboard interrupt...")
+            print("Stats:")
+            duration = time.time() - start
+            print("  Duration: {0}s".format(round(duration, 3)))
+            print("  Requests: {0} ({1} per minute)".format(
+                stats['total_requests'],
+                round(stats['total_requests'] / (duration / 60), 3)))
+            print("  Responses:", stats['total_responses'])
+            for status_code, count in stats['responses'].items():
+                print("   - {0}: {1}".format(status_code, count))
+            print("  Score requests:")
+            print("   - errored:", stats['score_requests_errored'])
+            print("   - returned:", stats['score_requests_returned'])
+            print("  Model scores:")
+            print("   - successes:")
+            for name, count in stats['model_scored'].items():
+                print("     - {0}: {1}".format(name, count))
+            print("   - errored:")
+            for name, type_doc in stats['model_errored'].items():
+                print("     - {0}:".format(name))
+                for type, count in type_doc.items():
+                    print("       - {0}: {1}".format(type, count))
+
+
+def request_and_stat(ores_url, context, models, rev_ids, stats, verbose):
+    path = "/v3/scores/{0}/".format(context)
+    logger.debug("Requesting {0} rev_ids...".format(len(rev_ids)))
+    params = {'revids': "|".join(str(r) for r in rev_ids),
+              'features': ""}
+    if models:
+        params = {'models': "|".join(models)}
+
+    stats['total_requests'] += 1
+    try:
+        response = requests.get(
+            ores_url + path, params=params)
+    except:
+        stats['responses']['connection error'] += 1
+        return
+
+    logger.debug("{0} response for {1} rev_ids...".format(
+        response.status_code, len(rev_ids)))
+    stats['total_responses'] += 1
+    stats['responses'][response.status_code] += 1
+
+    if response.status_code >= 200 and response.status_code < 300:
+        sys.stderr.write(".")
+        sys.stderr.flush()
```

Comment:
I would skip the flush, or only flush every 0=n mod 100.  Printing to console is incredibly expensive and will probably slow down the script.  Not sure how this will interact across threads, either...  Probably safer to do this with a https://docs.python.org/3/library/queue.html ?